src/components/tipRecords/TipPreview.vue

```diff
@@ -0,0 +1,526 @@
+      v-else
+      class="tip__article__content"
+    >
+      <div class="tip__no-preview">
+        <div class="source">
+          {{ sourceUrl }}
+        </div>
+        <div>{{ noPreview }}</div>
+      </div>
+      <div class="tip__links">
+        <div
+          class="tip__amount"
+          @click.stop
+        >
+          <TipInput :tip="tip" />
+        </div>
+        <div
+          class="site__url"
+          :title="tip.url"
+        >
+          <a
+            class="text-ellipsis"
+            target="_blank"
+            :href="tipUrl"
+            @click.stop
+          >
+            <ExternalLink />
+            <span class="text-ellipsis">{{ tip.url }}</span>
+          </a>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import Backend from '../../utils/backend';
+import TipInput from '../TipInput.vue';
+import YouTubeEmbed from './YouTubeEmbed.vue';
+import TwitterEmbed from './TwitterEmbed.vue';
+import SoundCloudEmbed from './SoundCloudEmbed.vue';
+import ExternalLink from '../../assets/externalLink.svg?icon-component';
+
+export default {
+  components: {
+    TipInput,
+    ExternalLink,
+    YouTubeEmbed,
+    TwitterEmbed,
+    SoundCloudEmbed,
+  },
+  props: {
+    tip: { type: Object, required: true },
+    goToTip: { type: Function, default: () => {} },
+    tipUrl: { type: String, default: '' },
+  },
+  data() {
+    return {
+      noPreview: 'No preview available for this URL',
+    };
+  },
+  computed: {
+    sourceUrl() {
+      const matches = this.tip.url.match(/^(https?:\/\/)?(www.)?([^/?#]+)(?:[/?#]|$)/i);
+      return matches ? matches[3] : '';
+    },
+    tipPreviewDescription() {
+      if (!this.isPreviewToBeVisualized(this.tip)) return '';
+
+      return this.tip.preview.description ? this.tip.preview.description : '';
+    },
+    tipPreviewTitle() {
+      if (!this.isPreviewToBeVisualized(this.tip)) return '';
+
+      return this.tip.preview.title ? this.tip.preview.title : '';
+    },
+    tipPreviewImage() {
+      return this.isPreviewToBeVisualized(this.tip) && this.tip.preview.image !== null ? Backend.getTipPreviewUrl(this.tip.preview.image) : '';
+    },
+  },
+  methods: {
+    isRichPreview(tip) {
+      const youTubeRegex = /^(https?:\/\/)?(www\.)?(youtube\.com|youtu\.?be)\/(watch|([a-zA-Z0-9_-]{11})).*$/;
+      const twitterRegex = /^(https?:\/\/)?(www\.)?twitter\.com\//;
+      const soundCloudRegex = /^(https?:\/\/)?(www\.)?soundcloud\.com\/[a-zA-Z0-9-_/]*$/;
+      return [
+        youTubeRegex,
+        twitterRegex,
+        soundCloudRegex]
+        .filter((r) => r.test(tip.url)).length > 0;
+    },
+    isPreviewToBeVisualized(tip) {
+      return typeof tip !== 'undefined' && tip !== null
+        && typeof tip.preview !== 'undefined'
+        && (
+          (tip.preview.description !== null && tip.preview.description.length > 0)
+          || (tip.preview.title !== null && tip.preview.title.length > 0)
+        );
+    },
```

Comment:
I would make it a computed variable:
```suggestion
    isPreviewToBeVisualized() {
      return this.tip.preview && (this.tip.preview.description || this.tip.preview.title);
    },
```